### PR TITLE
util: rename the encrypt spilled file config item to `data-encryption-method`

### PR DESCRIPTION
**Pull request by mmyj**
### What problem does this PR solve?

util: rename the encrypt spilled file config item to `data-encryption-method`

Issue Number: https://github.com/pingcap/tidb/issues/19388

Problem Summary:

We should keep the configurations of encryption are same as the TiKV.

### Release note 

rename the encrypt spilled file config item to `data-encryption-method`

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ allow-expression-index = true
 [isolation-read]
 engines = ["tiflash"]
 [security]
-spilled-file-encryption-method = "plaintext"
+data-encryption-method = "plaintext"
```
Comment (zz-jason) [inline config/config_test.go:215]: what's the config value in TiKV and PD which indicates no encryption?

Comment (XuHuaiyu) [reply config/config_test.go:215]: It's the same as TiKV
https://github.com/tikv/tikv/blob/4547ac41e784ec7a9a119a5fd0d83140ae565d57/etc/config-template.toml#L804

**Review by XuHuaiyu — COMMENTED** [review]
LGTM

File: config/config.toml.example [hunk 2]
```diff
@@ -191,7 +191,7 @@ cluster-ssl-key = ""
 # Configurations of the encryption method to use for encrypting the spilled data files.
 # Possible values are "plaintext", "aes128-ctr", if not set, it will be "plaintext" by default.
 # "plaintext" means encryption is disabled.
-spilled-file-encryption-method = "plaintext"
+data-encryption-method = "plaintext"
```
Comment (SunRunAway) [inline config/config.toml.example:194]: If possible, we can reuse the config `cluster-ssl-ca`, if it is on, then the spilled data files are encrypted.

Comment (SunRunAway) [reply config/config.toml.example:194]: cc @XuHuaiyu

**Comment by XuHuaiyu** [comment]
After the discussion, we'll keep the old configuration itme.

**Pull request closed without merging**